tools/toolchains/arm.py

```diff
@@ -284,8 +285,9 @@ def link(self, output, objects, libraries, lib_dirs, scatter_file):
         if lib_dirs:
             args.extend(["--userlibpath", ",".join(lib_dirs)])
         if scatter_file:
-            new_scatter = self.correct_scatter_shebang(scatter_file)
-            args.extend(["--scatter", new_scatter])
+            scatter_name = dirname(scatter_file)
```

Comment:
Oh yeah, my recommendation was really wrong here. This should probably be:
```suggestion
            scatter_name = relpath(scatter_file)
```

Reply:
Changes made.